Tasks/ANT/anttask.ts

```diff
@@ -105,20 +121,15 @@ async function doWork() {
         buildProps['sourcedirectories'] = sourceDirectories;
         buildProps['summaryfile'] = summaryFileName;
         buildProps['reportdirectory'] = reportDirectory;
-        buildProps['ccreporttask'] = ccReportTask
+        buildProps['ccreporttask'] = "CodeCoverage_9064e1d0"
         buildProps['reportbuildfile'] = reportBuildFile;
-        try {
-            var codeCoverageEnabler = new tl.CodeCoverageEnabler('Ant', ccTool);
-            codeCoverageEnabler.enableCodeCoverage(buildProps);
-            tl.debug("Code coverage is successfully enabled.");
-        }
-        catch (Error) {
-            tl.warning("Enabling code coverage failed. Check the build logs for errors.");
-        }
+
+        let ccEnabler = new CodeCoverageEnablerFactory().getTool("ant", ccTool.toLowerCase());
+        return ccEnabler.enableCodeCoverage(buildProps);
     }
 
     function publishCodeCoverage(codeCoverageOpted: boolean) {
-        if (codeCoverageOpted) {
+        if (codeCoverageOpted && ccReportTask) {
```

Comment:
ccReportTask is exists in Global level with async function and this function is called from that. apparently, the variable is available. to be clear, let me pass it as argument. will fix that